connectors/sources/network_drive.py

```diff
@@ -232,6 +233,26 @@ def fetch_members(self, group_name):
         return self.parse_output(members)
 
 
+class SMBSession(metaclass=Singleton):
+    _connection = None
+
+    def __init__(self, server_ip, username, password, port):
+        self.server_ip = server_ip
+        self.username = username
+        self.password = password
+        self.port = port
+        self.session = None
+
+    def create_connection(self):
+        """Creates an SMB session to the shared drive."""
+        self.session = smbclient.register_session(
+            server=self.server_ip,
+            username=self.username,
+            password=self.password,
+            port=self.port,
+        )
```

Comment:
Do you need to close it somehow btw?

Reply:
the connection is closed via `delete_session` method being called in close() method in NASDataSource class. We can move the actual logic in this class within a method `delete_connection()` and call that method inside close() of source class